src/utils/train.py

```diff
@@ -188,17 +194,32 @@ def evaluate_model(
     ground_truth = torch.cat(ground_truth).numpy()
     predictions = torch.cat(predictions).numpy()
 
-    average = "binary" if num_classes == 2 else "macro"
+    if no_average:
+        average = None
+    else:
+        average = "binary" if num_classes == 2 else "macro"
     precision, recall, f1_score, _ = precision_recall_fscore_support(ground_truth, predictions, average=average)
     accuracy = accuracy_score(ground_truth, predictions)
     result = {"accuracy": accuracy, "precision": precision, "recall": recall, "f1": f1_score}
 
     if group != "":
         result = {f"{group}/{k}": v for k, v in result.items()}
 
-    if log:
-        wandb.log(result)
+    if no_average:
+        columns = ["class_name"] + [k for k, v in result.items() if not k.endswith("accuracy")]
```

Comment:
```python
.. if k != "accuracy"
```

Reply:
but if group != "", keys will be like "test/accuracy", "val/recall"

Reply:
Since we are adding group in `log_dict`, you can just remove adding them into `result`. We only need group for correct logging